### PR TITLE
PERF: Fixed perf regression in TimedeltaIndex.get_loc

### LINKED ISSUE #34510: Performance regression in TimedeltaIndexing.time_get_loc
Setup:

```python
index = pd.timedelta_range(start="1985", periods=1000, freq="D")
timedelta = index[500]

%timeit index.get_loc(timedelta)
```

```
# master
7.39 µs ± 78.9 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

# 1.0.3
2.82 µs ± 13.5 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

See https://pandas.pydata.org/speed/pandas/#timedelta.TimedeltaIndexing.time_get_loc?commits=9929fca8-62c7dd3e

Happened somewhere between May 11 and May 27 (the gap in the benchmarks)

### PR DESCRIPTION
**Pull request by TomAugspurger**
Closes https://github.com/pandas-dev/pandas/issues/34510

1.0.3

```python
In [10]: index = pd.timedelta_range(start="1985", periods=1000, freq="D")
    ...: timedelta = index[500]
    ...:
    ...: %timeit index.get_loc(timedelta)

3.7 µs ± 275 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

```

This PR

```
3.25 µs ± 67.7 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

### CODE REVIEW

**Review by jreback — APPROVED** [review]
lgtm. @jbrockmendel if any comments

**Comment by jbrockmendel** [comment]
LGTM

File: doc/source/whatsnew/v1.1.0.rst [hunk 1]
```diff
@@ -760,6 +760,7 @@ Deprecations
 Performance improvements
 ~~~~~~~~~~~~~~~~~~~~~~~~
 
+- Fixed performance regression in :meth:`TimedeltaIndex.get_loc` (:issue:`34510`)
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v1.1.0.rst:763]: No whatsnew is needed, I think, the regression was only in master

**Comment by jreback** [comment]
lgtm

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Thanks!

**Pull request merged**